--- a/golden/beagle/graph.bgl
+++ b/golden/beagle/graph.bgl
@@ -168,7 +168,7 @@
                             (fn [(tid : String)]
                               (let [task (get task-map tid)]
                                 (if (some? task)
-                                  [(task-priority task) tid]
+                                  [(- 0 (task-priority task)) tid]
                                   [999999 tid])))
                             (filterv (fn [(tid : String)]
                                       (= (get true-in-degrees tid 0) 0))
@@ -203,7 +203,7 @@
                               (fn [(tid : String)]
                                 (let [task (get task-map tid)]
                                   (if (some? task)
-                                    [(task-priority task) tid]
+                                    [(- 0 (task-priority task)) tid]
                                     [999999 tid])))
                               (into rest-ready newly-ready)))]
             (recur all-ready new-result updated)))))))
